### PR TITLE
Stm32 clock ctrl common pll alternate clock sources + related fixes

### LINKED ISSUE #47101: drivers: clock_control: stm32 common: PLL_Q divider not converted to reg val
**Describe the bug**
According to the dt binding description the `div-q` device tree property defines the division factor for the PLL_Q output.
But the stm32_ll_common driver directly writes this value to the `RCC->PLLCFG` register without converting the division factor to the matching register value.

**To Reproduce**
N/A

**Expected behavior**
The value specified in dts should be converted to the register value that sets up the matching division factor.
i.e. shifting the value to the correct position of the register is not enough.

**Impact**
The PLL_Q clock source might be configured with a different frequency then intended.
Applies to STM32F4, STM32F7, STM32G0, STM32G4, STM32L4, STM32L5, STM32WB, STM32WL.

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Zephyr main (a1ef368094)

### PR DESCRIPTION
**Pull request by str4t0m**
This PR allows select STM32 SOCs to configure the PLL_P, PLL_Q, PLL_R outputs as alternate clock source.
The PLL outputs are enabled during initialization in case the respective divider is defined in dts.

The functionality is verified by extending the stm32_common_devices test with a test case that configures an alternate clock source for ADC1. Test configurations for STMG0, STM32G4, and STM32WL were changed accordingly.

Additionally, some related testcase and driver fixes are included:
- tests: drivers: clock_control: stm32_wl fix external clock dts
- drivers/clock_control: fix stm32 common pll_q divider setup
  Closes #47101 
- tests/drivers/clock_control: stm32_common_devices: lptim check disabled

Tested only on G0, G4, L4, WB, and WL targets.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for these fixes and improvements.
A couple of comments, otherwise LGTM

File: include/zephyr/drivers/clock_control/stm32_clock_control.h [hunk 1]
```diff
@@ -114,6 +114,7 @@
 	DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(pll), st_stm32l4_pll_clock, okay) || \
 	DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(pll), st_stm32u5_pll_clock, okay) || \
 	DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(pll), st_stm32wb_pll_clock, okay) || \
+	DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(pll), st_stm32wl_pll_clock, okay) || \
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply include/zephyr/drivers/clock_control/stm32_clock_control.h]: My mistake, removed.

File: include/zephyr/dt-bindings/clock/stm32g4_clock.h [hunk 2]
```diff
@@ -30,8 +30,11 @@
 #define STM32_SRC_SYSCLK	0x006
 /** Bus clock */
 #define STM32_SRC_PCLK		0x007
-/** PLL clocks */
-#define STM32_SRC_PLLCLK	0x008
+/** PLL clock outputs */
+#define STM32_SRC_PLL_P		0x008
+#define STM32_SRC_PLL_Q		0x009
+#define STM32_SRC_PLL_R		0x00a
+#define STM32_SRC_PLLCLK	0x00b
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply include/zephyr/dt-bindings/clock/stm32g4_clock.h]: Not really.
I thought about removing it, but wasn't sure so I kept it initially. PLLCLK is now removed on these series.
As a result, get_pllout_frequency() is no longer needed on those series, and therefore also removed.

Also added a commit to remove STM32_SRC_CLOCK_MIN/STM32_SRC_CLOCK_MAX.

**Review by erwango — APPROVED** [review]
Great work!

File: include/zephyr/dt-bindings/clock/stm32g4_clock.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (erwango) [reply include/zephyr/dt-bindings/clock/stm32g4_clock.h]: good, thanks

**Review by carlescufi — APPROVED** [review]
Based on the header changes

**Pull request merged**